app/controllers/peeping.rb

```diff
@@ -0,0 +1,15 @@
+class Chitter < Sinatra::Base
+  get '/' do
+    @peeps = Peep.all(:order => [:created_at.desc])
+    erb :index
+  end
+
+  post '/peeps' do
+    @user = current_user
+    Peep.create(message: params[:peep], user_id: @user.id)
+    User.all.each do |user|
```

Comment:
The way I read this code, you are pulling all users from the database, and then if the text of the peep contains their username they are being sent an alert - is that correct?

This would probably be a problem if you had lots of users in your database, because you'd have to get each record, transform it into a Ruby instance, and then do a fuzzy text match. Better would be if you could identify user names from the text, e.g. with an @ symbol, and then do a User.find(:name => username) for each identified name.